src/blackbook/__init__.py

```diff
@@ -27,12 +27,17 @@ def format_notebook_content(path: pathlib.Path) -> Optional[dict]:
 
                 try:  # Some ipynb files will not have valid source code
                     string = "".join(cell["source"])
-                    formatted_string = black.format_str(string, mode=black.FileMode())
+                    formatted_string = black.format_str(
+                        string, mode=black.FileMode()
+                    ).strip()  # Remove trailing newlines
                     if formatted_string != string:
                         modification_found = True
+                        cell_lines = formatted_string.splitlines()
+                        n_last_newline = len(cell_lines) - 1
```

Comment:
These two lines are added only so that they are precomputed for the list comprehension, and to make it (hopefully) easier to read than

```python
cell["source"] = [
    line + "\n" if itr < len(formatted_string.splitlines()) - 1 else line
    for itr, line in enumerate(formatted_string.splitlines())
]
```